### PR TITLE
Don't swallow FileNotFoundError in ExperimentList.from_file

### LINKED ISSUE #154: ExperimentList.from_file fails obscurely when image data are missing
```
>>> from dxtbx.model import ExperimentList
>>> el=ExperimentList.from_file('indexed.expt')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/fcx32934/sw/cctbx/modules/dxtbx/model/__init__.py", line 846, in from_file
    return ExperimentListFactory.from_serialized_format(filename, check_format)
  File "/home/fcx32934/sw/cctbx/modules/dxtbx/model/experiment_list.py", line 750, in from_serialized_format
    return ExperimentListFactory.from_pickle_file(filename)
  File "/home/fcx32934/sw/cctbx/modules/dxtbx/model/experiment_list.py", line 714, in from_pickle_file
    obj = pickle.load(infile)
_pickle.UnpicklingError: invalid load key, '{'.
```
seems to be for pickles only?

### PR DESCRIPTION
**Pull request by rjgildea**
Otherwise this confusingly leads to an error under from_pickle_file.
Fixes #154

### CODE REVIEW

**Comment by dagewa** [comment]
I think this is Python 3 only

**Comment by dagewa** [comment]
Perhaps we need to catch `Exception as e` then check if it was `FileNotFoundError` under an `if six.PY3`?

File: model/experiment_list.py [hunk 1]
```diff
@@ -743,6 +743,8 @@ def from_serialized_format(filename, check_format=True):
         # First try as a JSON file
         try:
             return ExperimentListFactory.from_json_file(filename, check_format)
+        except FileNotFoundError:
```
Comment (Anthchirp) [inline model/experiment_list.py]:
```suggestion
        except OSError:
```

File: tests/model/test_experiment_list.py [hunk 2]
```diff
@@ -799,7 +799,25 @@ def compare_experiment(exp1, exp2):
 
 
 def test_experimentlist_from_file(dials_regression, tmpdir):
-    # This allows expansion of environment variables in regression files
+    # With the default check_format=True this file should fail to load with an
+    # appropriate error as we can't find the images on disk
+    with pytest.raises(FileNotFoundError) as e:
```
Comment (Anthchirp) [inline tests/model/test_experiment_list.py]:
```suggestion
    with pytest.raises(OSError) as e:
```

**Comment by Anthchirp** [comment]
if you wanted to make sure you get the correct error reason then comparing `e.errno` to `errno.ENOENT` is probably cleaner

File: model/experiment_list.py (hunk 1 above)
Comment (rjgildea) [reply model/experiment_list.py]: It looks like it's actually `IOError` not `OSError`, at least the test now passes for me on both Python 2 and Python 3.

File: model/experiment_list.py [hunk 3]
```diff
@@ -743,6 +744,11 @@ def from_serialized_format(filename, check_format=True):
         # First try as a JSON file
         try:
             return ExperimentListFactory.from_json_file(filename, check_format)
+        except IOError as e:
+            # In an ideal Python 3 world this would be much more simply FileNotFoundError
+            if e.errno == errno.ENOENT:
+                raise
+            pass
```
Comment (Anthchirp) [inline model/experiment_list.py]:
```suggestion
```

redundant

**Pull request merged**